js/client-chat.js

```diff
@@ -569,9 +569,18 @@
 					}
 					switch (targets[0]) {
 					case 'add':
+						var test;
 						for (var i=1, len=targets.length; i<len; i++) {
-							highlights.push(targets[i].trim());
+							if (/[\\^$*+?()|{}[\]]/.test(targets[i])) {
+								// Catch any errors thrown by newly added regular expressions so they don't break the entire highlight list
+								try {
+									test = new RegExp(targets[i]);
```

Comment:
We don't need a var test; we can just `new RegExp(targets[i]);`

Even if we needed the var test, it should be declared here, not at the top of the case.